src/components/ArticleDetail.jsx

```diff
@@ -0,0 +1,40 @@
+import { useParams } from "react-router-dom";
+import { useEffect, useState } from "react";
+import { fetchArticleById } from "../utils/api";
+
+function ArticleDetail() {
+  const { article_id } = useParams();
+  const [article, setArticle] = useState(null);
+
+  useEffect(() => {
+    fetchArticleById(article_id)
+      .then((fetchedArticle) => {
+        setArticle(fetchedArticle);
+      })
+      .catch((error) => {
+        console.error("Error fetching article:", error);
+      });
+  }, [article_id]);
```

Comment:
Every time the url changes for the endpoint /articles/:article_id then the ArticleDetail component will re-render.
As it re-renders every time you change the url you don't need article_id in the dependency array of the useEffect as the useEffect runs on every re-render by default